Support Postgres 17 in PostgreSQLDatabaseMetrics

The-East-Wind opened the pull request:
Resolves #6657 

Stats removed in Pg 17 and their alternatives:

| < Pg17 | Pg17+|
|:---:|:---:|
|`pg_stat_bgwriter.checkpoints_timed`|`pg_stat_checkpointer.num_timed`|
|`pg_stat_bgwriter.checkpoints_req`|`pg_stat_checkpointer.num_requested`|
|`pg_stat_bgwriter.buffers_backend`|`SUM(pg_stat_io.writes)`|
|`pg_stat_bgwriter.buffers_checkpoint`|`pg_stat_checkpointer.buffers_written`|

Reference: https://www.postgresql.org/docs/17/monitoring-stats.html

shakuzen reviewed (commented):
The general approach here looks good to me. I'm not sure if @jonatan-ivanov has any thoughts on it or not.

shakuzen commented on micrometer-core/src/main/java/io/micrometer/core/instrument/binder/db/PostgreSQLDatabaseMetrics.java:
```diff
@@ -366,4 +408,59 @@ private Names() {
+            try {
+                final Matcher matcher = VERSION_PATTERN.matcher(versionString);
+                if (!matcher.matches()) {
+                    return EMPTY;
```
Are there expected cases in which the version string the server returns won't be parseable by this method? I'm wondering if we should log something when parsing the version fails, or perhaps when `getServerVersion` returns `Version.EMPTY`. Otherwise, users may not notice the issue early on and it will delay feedback about a situation we potentially need to fix this code to handle. What do you think?

shakuzen reviewed (approved):
This looks good to me. I left one follow-up comment. We could also handle this in a polish commit when merging.

The-East-Wind commented:
> This looks good to me. I left one follow-up comment. We could also handle this in a polish commit when merging.

I've addressed the comment, check whenever you are free and let me know if anything else needs to be changed or added. Thank you.
